### ISSUE TITLE
Markdown syntax highlighting stops working after `default` keyword is used in TS code block

### ISSUE THREAD

**Issue by joeskeen** [Microsoft/vscode#8903]
- VSCode Version: 1.3.0-insider
- OS Version: OS X El Capitan

Steps to Reproduce:
1. Create a new Markdown file
2. Paste in this content (taken from the TypeScript Handbook):

<pre>
##### JQuery.d.ts

```ts
declare let $: JQuery;
export default $;
```

##### App.ts

```ts
import $ from "JQuery";

$("button.continue").html( "Next Step..." );
```

Classes and function declarations can be authored directly as default exports.
Default export class and function declaration names are optional.
</pre>

1. You will notice that after the first code block the heading isn't the right color, as if it was treating it as a template string in the TypeScript code.
   ![image](https://cloud.githubusercontent.com/assets/12286274/16679876/4b5b72da-44a8-11e6-8b3a-270cb415057e.png)
2. Remove the word `default` from the first code block
3. Everything is highlighted properly
   ![image](https://cloud.githubusercontent.com/assets/12286274/16679882/59c5f1ec-44a8-11e6-8365-45e535aa9f96.png)

**Comment by mjbvz** [comment]
Fixed by #14626

![screen shot 2016-11-10 at 3 58 55 pm](https://cloud.githubusercontent.com/assets/12821956/20199330/9c07d2b4-a75e-11e6-84d3-466d0657c147.png)

**Comment by joeskeen** [comment]
Thanks!
